editor/src/templates/editor.tsx

```diff
@@ -227,120 +255,62 @@ export class Editor {
-                            true,
-                            this.storedState.editor.vscodeBridgeReady,
-                          ),
-                      )
-                    } else {
-                      renderProjectLoadError(importProjectResult.errorMessage)
-                    }
-                  })
-                  .catch((err) => {
-                    console.error('Import error.', err)
-                  })
-              }
-            })
-          } else if (githubOwner != null && githubRepo != null) {
-            renderProjectLoadError('Github repo import is only supported for logged in users')
-          } else {
-            createNewProject(this.boundDispatch, () =>
-              renderRootComponent(
-                this.utopiaStoreHook,
-                this.utopiaStoreApi,
-                this.spyCollector,
-                true,
-                this.storedState.editor.vscodeBridgeReady,
-              ),
-            )
-          }
-        } else if (isCookiesOrLocalForageUnavailable(loginState)) {
-          createNewProject(this.boundDispatch, () =>
-            renderRootComponent(
-              this.utopiaStoreHook,
-              this.utopiaStoreApi,
-              this.spyCollector,
-              true,
-              this.storedState.editor.vscodeBridgeReady,
-            ),
-          )
+        if (isLoggedIn(loginState)) {
+          this.storedState.persistence.login()
+        }
+
+        if (isCookiesOrLocalForageUnavailable(loginState) || projectId == null) {
+          this.storedState.persistence.createNew(createNewProjectName(), defaultProject())
         } else {
-          projectIsStoredLocally(projectId).then((isLocal) => {
-            if (isLocal) {
-              loadFromLocalStorage(
-                projectId,
-                this.boundDispatch,
-                isLoggedIn(loginState),
-                this.storedState.workers,
-                () =>
-                  renderRootComponent(
-                    this.utopiaStoreHook,
-                    this.utopiaStoreApi,
-                    this.spyCollector,
-                    true,
-                    this.storedState.editor.vscodeBridgeReady,
-                  ),
-              )
-            } else {
-              loadFromServer(
-                projectId,
-                this.boundDispatch,
-                this.storedState.workers,
-                () => {
-                  renderRootComponent(
-                    this.utopiaStoreHook,
-                    this.utopiaStoreApi,
-                    this.spyCollector,
-                    true,
-                    this.storedState.editor.vscodeBridgeReady,
-                  )
-                },
-                () => {
-                  renderProjectNotFound()
-                },
-              )
-            }
-          })
+          this.storedState.persistence.load(projectId)
         }
+
+        // FIXME Restore github project importing
```

Comment:
I'm fixing this up now, so this will either be fixed in time for the PR to be merged, or in an immediate follow up